PreMailer.Net/PreMailer.Net/CssElementStyleResolver.cs

```diff
@@ -8,19 +8,51 @@ public static class CssElementStyleResolver
 	{
 		private const string premailerAttributePrefix = "-premailer-";
 
-		public static IEnumerable<AttributeToCss> GetAllStyles(IElement domElement, StyleClass styleClass)
+	public static IEnumerable<AttributeToCss> GetAllStyles(IElement domElement, StyleClass styleClass)
+	{
+		var attributeCssList = new List<AttributeToCss>();
+		var originalStyleAttr = domElement.Attributes["style"];
+		
+		AddSpecialPremailerAttributes(attributeCssList, styleClass);
+		
+		if (originalStyleAttr != null)
 		{
-			var attributeCssList = new List<AttributeToCss>();
-
-			AddSpecialPremailerAttributes(attributeCssList, styleClass);
-
+			var parser = new CssParser();
+			var originalStyleClass = parser.ParseStyleClass("inline", originalStyleAttr.Value);
+			
+			var mergedStyleClass = new StyleClass();
+			
 			if (styleClass.Attributes.Count > 0)
-				attributeCssList.Add(new AttributeToCss { AttributeName = "style", CssValue = styleClass.ToString() });
-
-			attributeCssList.AddRange(CssStyleEquivalence.FindEquivalent(domElement, styleClass));
-
-			return attributeCssList;
+			{
+				mergedStyleClass.Merge(styleClass, true);
+			}
+			
+			foreach (var attr in originalStyleClass.Attributes)
+			{
+				if (attr.Important)
+				{
+					mergedStyleClass.Attributes.Merge(attr);
+				}
+				else if (!mergedStyleClass.Attributes.TryGetValue(attr.Style, out var existing) || !existing.Important)
+				{
+					mergedStyleClass.Attributes.Merge(attr);
+				}
+			}
+			
+			if (mergedStyleClass.Attributes.Count > 0)
+			{
+				attributeCssList.Add(new AttributeToCss { AttributeName = "style", CssValue = mergedStyleClass.ToString(emitImportant: true) });
+			}
+		}
+		else if (styleClass.Attributes.Count > 0)
+		{
+			attributeCssList.Add(new AttributeToCss { AttributeName = "style", CssValue = styleClass.ToString(emitImportant: true) });
```

Comment:
seems to me this could be done more elegantly, as adding to the list is exactly the same except where it comes from. Perhaps it is possible to avoid the `else if` block?